### ISSUE TITLE
Installing orjson `3.10.0` does not include maturin

### ISSUE THREAD

**Issue by MarkusSagen** [ijl/orjson#465]
When installing the latest version of orjson (`3.10.0`) with poetry, I get the following issues. 

This does not happen when installing >= `3.9.15`

```

Package operations: 2 installs, 2 updates, 0 removals

  • Downgrading packaging (24.0 -> 23.2)
  • Downgrading anyio (4.3.0 -> 3.7.1)
  • Installing orjson (3.10.0): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel
  
  Running `maturin pep517 build-wheel -i /var/folders/7l/0vdyz7nn5vn9zn11g6xtx06c0000gn/T/tmpbe4f8ndr/.venv/bin/python --compatibility off`
  Traceback (most recent call last):
    File "/Users/markusjohnson/.pyenv/versions/3.10.10/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
      main()
    File "/Users/markusjohnson/.pyenv/versions/3.10.10/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/Users/markusjohnson/.pyenv/versions/3.10.10/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
      return _build_backend().build_wheel(wheel_directory, config_settings,
    File "/var/folders/7l/0vdyz7nn5vn9zn11g6xtx06c0000gn/T/tmpbe4f8ndr/.venv/lib/python3.10/site-packages/maturin/__init__.py", line 108, in build_wheel
      return _build_wheel(wheel_directory, config_settings, metadata_directory)
    File "/var/folders/7l/0vdyz7nn5vn9zn11g6xtx06c0000gn/T/tmpbe4f8ndr/.venv/lib/python3.10/site-packages/maturin/__init__.py", line 89, in _build_wheel
      result = subprocess.run(command, stdout=subprocess.PIPE)
    File "/Users/markusjohnson/.pyenv/versions/3.10.10/lib/python3.10/subprocess.py", line 503, in run
      with Popen(*popenargs, **kwargs) as process:
    File "/Users/markusjohnson/.pyenv/versions/3.10.10/lib/python3.10/subprocess.py", line 971, in __init__
      self._execute_child(args, executable, preexec_fn, close_fds,
    File "/Users/markusjohnson/.pyenv/versions/3.10.10/lib/python3.10/subprocess.py", line 1847, in _execute_child
      raise child_exception_type(errno_num, err_msg, err_filename)
  FileNotFoundError: [Errno 2] No such file or directory: 'maturin'
  

  at ~/.pyenv/versions/3.10.10/lib/python3.10/site-packages/poetry/installation/chef.py:166 in _prepare
      162│ 
      163│                 error = ChefBuildError("\n\n".join(message_parts))
      164│ 
      165│             if error is not None:
    → 166│                 raise error from None
      167│ 
      168│             return path
      169│ 
      170│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with orjson (3.10.0) not supporting PEP 517 builds. You can verify this by running 'pip wheel --no-cache-dir --use-pep517 "orjson (==3.10.0)"'.
```

---

Saw that the new version vas deployed recently, so might have been me installing at the wrong time 😅

**Comment by messense** [comment]
This is a Poetry bug: https://github.com/python-poetry/poetry/issues/8749, it has nothing to do with orjson/maturin, consider upgrade to latest Poetry version.

**Comment by DbCrWk** [comment]
@messense, I don't believe this is a `poetry` issue. I'm reposting what I put in #466.

To verify, I ran `pip wheel --no-cache-dir --use-pep517 "orjson (==3.10.0)"` and the following output:

```
Collecting orjson==3.10.0
  Downloading orjson-3.10.0.tar.gz (4.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.9/4.9 MB 43.4 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
      
      Checking for Rust toolchain....
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by DbCrWk** [comment]
I installed rust and put it in my `PATH` and that did solve the issue for now:
https://www.rust-lang.org/tools/install

**Comment by bigpick** [comment]
I don't think its that either - I have `rustc` and `cargo` both in my `PATH` and it still fails when trying to use poetry.

OTOH, using the `pip wheel --no-cache-dir --use-pep517 "orjson (==3.10.0)"` command works:

```
pip wheel --no-cache-dir --use-pep517 "orjson (==3.10.0)"
Collecting orjson==3.10.0
  Downloading orjson-3.10.0.tar.gz (4.9 MB)
...
Successfully built orjson
```

So it does seem to be something wrong specifically with how orjson doesn't support what poetry wants?

**Comment by bushwhackr** [comment]
TL;DR: macos wheels are not published when you first run a poetry lock.
Solution: Rerun poetry lock to retrieve the macos entries.

I found the root cause to be a missing entry for macos package when you first do a poetry lock. I looked at the release process and can't seem to find where macos wheels are being generated, hence the difference in the release artifacts depending on the time the pypi query was made.

```diff
--- poetry-old.lock     2024-04-01 10:32:22
+++ poetry-new.lock     2024-04-01 10:35:56
@@ -1,6 +1,7 @@
 [[package]]
 description = "Fast, correct Python JSON library supporting dataclasses, datetimes, and numpy"
 files = [
+    {file = "orjson-3.10.0-cp310-cp310-macosx_10_15_x86_64.macosx_11_0_arm64.macosx_10_15_universal2.whl", hash = "sha256:47af5d4b850a2d1328660661f0881b67fdbe712aea905dadd413bdea6f792c33"},
     {file = "orjson-3.10.0-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:c90681333619d78360d13840c7235fdaf01b2b129cb3a4f1647783b1971542b6"},
     {file = "orjson-3.10.0-cp310-cp310-manylinux_2_17_armv7l.manylinux2014_armv7l.whl", hash = "sha256:400c5b7c4222cb27b5059adf1fb12302eebcabf1978f33d0824aa5277ca899bd"},
     {file = "orjson-3.10.0-cp310-cp310-manylinux_2_17_ppc64le.manylinux2014_ppc64le.whl", hash = "sha256:5dcb32e949eae80fb335e63b90e5808b4b0f64e31476b3777707416b41682db5"},
@@ -28,6 +29,9 @@
     {file = "orjson-3.10.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:237ba922aef472761acd697eef77fef4831ab769a42e83c04ac91e9f9e08fa0e"},
     {file = "orjson-3.10.0-cp312-cp312-musllinux_1_2_aarch64.whl", hash = "sha256:98c1bfc6a9bec52bc8f0ab9b86cc0874b0299fccef3562b793c1576cf3abb570"},
     {file = "orjson-3.10.0-cp312-cp312-musllinux_1_2_x86_64.whl", hash = "sha256:30d795a24be16c03dca0c35ca8f9c8eaaa51e3342f2c162d327bd0225118794a"},
+    {file = "orjson-3.10.0-cp312-none-win32.whl", hash = "sha256:6a3f53dc650bc860eb26ec293dfb489b2f6ae1cbfc409a127b01229980e372f7"},
+    {file = "orjson-3.10.0-cp312-none-win_amd64.whl", hash = "sha256:983db1f87c371dc6ffc52931eb75f9fe17dc621273e43ce67bee407d3e5476e9"},
+    {file = "orjson-3.10.0-cp38-cp38-macosx_10_15_x86_64.macosx_11_0_arm64.macosx_10_15_universal2.whl", hash = "sha256:9a667769a96a72ca67237224a36faf57db0c82ab07d09c3aafc6f956196cfa1b"},
     {file = "orjson-3.10.0-cp38-cp38-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:ade1e21dfde1d37feee8cf6464c20a2f41fa46c8bcd5251e761903e46102dc6b"},
     {file = "orjson-3.10.0-cp38-cp38-manylinux_2_17_armv7l.manylinux2014_armv7l.whl", hash = "sha256:23c12bb4ced1c3308eff7ba5c63ef8f0edb3e4c43c026440247dd6c1c61cea4b"},
     {file = "orjson-3.10.0-cp38-cp38-manylinux_2_17_ppc64le.manylinux2014_ppc64le.whl", hash = "sha256:b2d014cf8d4dc9f03fc9f870de191a49a03b1bcda51f2a957943fb9fafe55aac"},
@@ -37,6 +41,7 @@
     {file = "orjson-3.10.0-cp38-cp38-musllinux_1_2_x86_64.whl", hash = "sha256:13b5d3c795b09a466ec9fcf0bd3ad7b85467d91a60113885df7b8d639a9d374b"},
     {file = "orjson-3.10.0-cp38-none-win32.whl", hash = "sha256:5d42768db6f2ce0162544845facb7c081e9364a5eb6d2ef06cd17f6050b048d8"},
     {file = "orjson-3.10.0-cp38-none-win_amd64.whl", hash = "sha256:33e6655a2542195d6fd9f850b428926559dee382f7a862dae92ca97fea03a5ad"},
+    {file = "orjson-3.10.0-cp39-cp39-macosx_10_15_x86_64.macosx_11_0_arm64.macosx_10_15_universal2.whl", hash = "sha256:4050920e831a49d8782a1720d3ca2f1c49b150953667eed6e5d63a62e80f46a2"},
     {file = "orjson-3.10.0-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:1897aa25a944cec774ce4a0e1c8e98fb50523e97366c637b7d0cddabc42e6643"},
     {file = "orjson-3.10.0-cp39-cp39-manylinux_2_17_armv7l.manylinux2014_armv7l.whl", hash = "sha256:9bf565a69e0082ea348c5657401acec3cbbb31564d89afebaee884614fba36b4"},
     {file = "orjson-3.10.0-cp39-cp39-manylinux_2_17_ppc64le.manylinux2014_ppc64le.whl", hash = "sha256:b6ebc17cfbbf741f5c1a888d1854354536f63d84bee537c9a7c0335791bb9009"},
```

**Comment by BedirT** [comment]
I don't think the issue is resolved. I am having the same issue. I have rust in path + I re-created the poetry lock and even re-created the environment. Nothing seems to be working. I am running the latest poetry `1.8.2` and pip `24.0`.

**Comment by bushwhackr** [comment]
@BedirT https://github.com/ijl/orjson/issues/466#issuecomment-2030773231
